reference_data/models.py

```diff
@@ -57,74 +58,61 @@ class HumanPhenotypeOntology(models.Model):
+    coding_region_size_grch38 = models.IntegerField(default=0)  # number of protein-coding base-pairs in this gene (= 0 for non-coding genes)
 
-    protein_id = models.CharField(max_length=20, null=True)
+    # gencode-specific fields, although models could hypothetically come from refseq or other places
+    gencode_gene_type = models.TextField(null=True, blank=True)
+    gencode_release = models.IntegerField(null=True, blank=True)
 
-    class Meta:
-        unique_together = ('gencode_release', 'chrom', 'start', 'end', 'gene_id')
 
+class TranscriptInfo(models.Model):
+    gene = models.ForeignKey(GeneInfo, on_delete=models.CASCADE)
 
-class GencodeTranscript(models.Model):
-    gencode_release = models.ForeignKey(GencodeRelease, on_delete=models.PROTECT)
-    gene = models.ForeignKey(GencodeGene, on_delete=models.CASCADE)
+    transcript_id = models.CharField(max_length=20, db_index=True, unique=True)  # without the version suffix
+    #protein_id = models.CharField(max_length=20, null=True)
 
-    chrom = models.CharField(max_length=1)
-    start = models.IntegerField()
-    end = models.IntegerField()
+    chrom_grch37 = models.CharField(max_length=2, null=True, blank=True)
+    start_grch37 = models.IntegerField(null=True, blank=True)
+    end_grch37 = models.IntegerField(null=True, blank=True)
+    strand_grch37 = models.CharField(max_length=1, null=True, blank=True)
+    coding_region_size_grch37 = models.IntegerField(default=0)  # number of protein-coding bases (= 0 for non-coding genes)
 
-    source = models.CharField(max_length=1, choices=GENCODE_SOURCE_CHOICES)
-    strand = models.CharField(max_length=1)
+    chrom_grch38 = models.CharField(max_length=2, null=True, blank=True)
+    start_grch38 = models.IntegerField(null=True, blank=True)
+    end_grch38 = models.IntegerField(null=True, blank=True)
+    strand_grch38 = models.CharField(max_length=1, null=True, blank=True)
+    coding_region_size_grch38 = models.IntegerField(default=0)  # number of protein-coding bases (= 0 for non-coding genes)
 
-    transcript_id = models.CharField(max_length=20, db_index=True)  # without the version suffix
-    transcript_status = models.CharField(max_length=1, choices=GENCODE_STATUS_CHOICES)
-    transcript_name = models.CharField(max_length=30, db_index=True)
 
-    transcript_support_level = models.IntegerField(null=True)
+# based on # ftp://ftp.broadinstitute.org/pub/ExAC_release/release0.3.1/functional_gene_constraint/fordist_cleaned_exac_r03_march16_z_pli_rec_null_data.txt
+class GeneConstraint(models.Model):
+    gene = models.ForeignKey(GeneInfo, on_delete=models.CASCADE)
 
-    class Meta:
-        unique_together = ('gencode_release', 'chrom', 'start', 'end', 'transcript_id')
+    mis_z = models.FloatField()
+    mis_z_rank = models.IntegerField()
+    pLI = models.FloatField()
+    pLI_rank = models.IntegerField()
+
+    #pRec = models.FloatField()
 
 
 class OMIM(models.Model):
```

Comment:
Can we make this `Omim`, instead of all caps?

Reply:
yes